WordPress/performance
# Add web worker offloading plugin

## Summary


Fixes https://github.com/WordPress/performance/issues/176

Previously: https://github.com/WordPress/performance/pull/556

## Relevant technical choices

- Add Web Worker Offloading standalone plugin which allows running scripts within the worker thread with the help of the Partytown library.
- Add the `wwo_configuration` filter which can be used to update web worker [configurations](https://partytown.builder.io/configuration).

### Testing steps
- Enable this performance plugin or use `wp-env` in which this plugin is enabled by default on this branch.
- Add a JS script with `web-worker-offloading` as a dependency. When the page is loaded, the JS script should be loaded within the web worker.

  ```php
  function enqueue_non_critical_js() {
	  wp_enqueue_script(
		  'non-critical-js',
		  '/path/to/non-critical.js',
		  array( 'web-worker-offloading' ), // Note the `web-worker-offloading` being added as a dep
		  '1.0.0',
		  true
	  );
  }
  add_action( 'wp_enqueue_scripts', 'enqueue_non_critical_js' );
  ```
- Use `wwo_configuration` hook to update web worker [configurations](https://partytown.builder.io/configuration).
    ```php
    function update_wwo_config( $config ) {
      $config["debug"] = true;
      return $config;
    }
    add_filter( 'wwo_configuration', 'update_wwo_config' );
    ```
